src/datastore/postgres-store.ts

```diff
@@ -1514,20 +1514,42 @@ export class PgDataStore
     });
   }
 
-  async resolveBnsNames(zonefile: string, atch_resolved: boolean, tx_id: string): Promise<void> {
+  async resolveBnsNames(
+    zonefile: string,
+    zonefile_hash: string,
+    atch_resolved: boolean,
+    tx_id: string
+  ): Promise<void> {
     await this.queryTx(async client => {
+      // inserting zonefile into zonefiles table
+      const validZonefileHash = this.validateZonefileHash(zonefile_hash);
+      await client.query(
+        `
+        UPDATE zonefiles
+        SET zonefile = $1
+        WHERE zonefile_hash = $2
+        `,
+        [zonefile, validZonefileHash]
+      );
       await client.query(
         `
         UPDATE names
-        SET zonefile = $1, atch_resolved = $2
-        WHERE tx_id = $3 AND canonical = true AND microblock_canonical = true
+        SET atch_resolved = $1
+        WHERE tx_id = $2 AND canonical = true AND microblock_canonical = true
         `,
-        [zonefile, atch_resolved, hexToBuffer(tx_id)]
+        [atch_resolved, hexToBuffer(tx_id)]
       );
     });
     this.emit('nameUpdate', tx_id);
   }
 
+  private validateZonefileHash(zonefileHash: string) {
```

Comment:
Created this function to defined a standard for zonefile hash such that it starts with `0x`.

Reply:
use `has0xPrefix` from `helpers.ts`